src/main/java/duke/commands/CommandEventHandler.java

```diff
@@ -0,0 +1,46 @@
+package duke.commands;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import duke.exceptions.DukeException;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.utils.DateTime;
+
+
+/**
+ * Handler for the Event command.
+ */
+public class CommandEventHandler extends CommandHandler {
+
+    public CommandEventHandler(String value, String flag, String additionalValue) throws DukeException {
+        super(value, flag, additionalValue);
+    }
+
+    /**
+     * Checks the validity of the syntax after being parsed by CommandParser.
+     * @throws DukeException error message.
+     */
+    public void checkValid() throws DukeException {
+        if (value == null || !flag.equals("at") || additionalValue == null) {
```

Comment:
Perhaps you could try to avoid complicated expressions here by initialising them first and use them in the if else statement
```suggestion
        boolean isValid = value == null || !flag.equals("at") || additionalValue == null;
        if (isValid) {
```

Reply:
Thanks for the suggestion!